Cython/Utility/CpdefEnums.pyx

```diff
@@ -6,7 +6,11 @@ cdef extern from *:
     int PY_VERSION_HEX
 
 cdef object __Pyx_OrderedDict
-from collections import OrderedDict as __Pyx_OrderedDict
+
+if PY_VERSION_HEX >= 0x03070000:
```

Comment:
So practically we only use `__Pyx_EnumMeta` in Python<3.4 (but it's defined at the top level because cdef classes have to be). Therefore I think `__Pyx_OrderedDict = dict` is never used and we could simplify this and only do the `OrderedDict` import for Python<3.4 (and leave it uninitialised otherwise)

Reply:
I've tried this, but it breaks some tests. It seems that `__Pyx_EnumMeta` is always instantiated.

<details>
<summary>log</summary>

```
Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]

Running tests against Cython 3.0.0a11 323bae936f0c0876d80c4f572523b7b42eb4d52c + uncommitted changes
Using Cython language level 3.
Test dependency not found: 'numpy'
Test dependency not found: 'pythran'
Test dependency found: 'setuptools.sandbox' version 65.6.3
Test dependency found: 'asyncio' version 3.10.9
Test dependency found: 'pstats' version 3.10.9
Test dependency not found: 'posix'
Test dependency found: 'array' version 3.10.9
Test dependency not found: 'Cython.Coverage'
Test dependency not found: 'Cython.Coverage'
Test dependency not found: 'IPython.testing.globalipapp'
Test dependency not found: 'jedi_BROKEN_AND_DISABLED'
Test dependency found: 'test.support' version 3.10.9
Backends: c,cpp

runTest (__main__.CythonRunTestCase)
[-1] compiling (c/cy3) and running cpdef_enums ... ERROR
runTest (__main__.CythonRunTestCase)
[-1] compiling (cpp/cy3) and running cpdef_enums ... ERROR
runTest (__main__.EndToEndTest)
[-1] End-to-end cpdef_enums_import ... [-1] ['C:\\Program Files\\Python310\\python.exe', 'setup.py', 'build_ext', '--inplace']
Compiling enums.pyx because it changed.
Compiling no_enums.pyx because it changed.
[1/2] Cythonizing enums.pyx
[2/2] Cythonizing no_enums.pyx

D:\Documents\git\cython\Cython\Compiler\Main.py:370: FutureWarning: Cython directive 'language_level' not set, using '3str' for now (Py3). This has changed from earlier releases! File: D:\Documents\git\cython\TEST_TMP\run\cpdef_enums_import\enums.pxd
  tree = Parsing.p_module(s, pxd, full_module_name)
D:\Documents\git\cython\Cython\Compiler\Main.py:370: FutureWarning: Cython directive 'language_level' not set, using '3str' for now (Py3). This has changed from earlier releases! File: D:\Documents\git\cython\TEST_TMP\run\cpdef_enums_import\no_enums.pyx
  tree = Parsing.p_module(s, pxd, full_module_name)


[-1] ['C:\\Program Files\\Python310\\python.exe', '-c', 'import import_enums_test']

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "D:\Documents\git\cython\TEST_TMP\run\cpdef_enums_import\import_enums_test.py", line 3, in <module>
    from enums import *
  File "<stringsource>", line 25, in init enums
  File "<stringsource>", line 17, in EnumBase.__Pyx_EnumMeta.__init__
TypeError: 'NoneType' object is not callable


Final directory layout of 'cpdef_enums_import':
.\enums.c
.\enums.cp310-win_amd64.pyd
.\enums.pxd
.\enums.pyx
.\import_enums_test.py
.\no_enums.c
.\no_enums.cp310-win_amd64.pyd
.\no_enums.pyx
.\setup.py
.\build\lib.win-amd64-cpython-310\enums.cp310-win_amd64.pyd
.\build\lib.win-amd64-cpython-310\no_enums.cp310-win_amd64.pyd
.\build\temp.win-amd64-cpython-310\Release\enums.cp310-win_amd64.def
.\build\temp.win-amd64-cpython-310\Release\enums.o
.\build\temp.win-amd64-cpython-310\Release\no_enums.cp310-win_amd64.def
.\build\temp.win-amd64-cpython-310\Release\no_enums.o
.\__pycache__\import_enums_test.cpython-310.pyc


#### 2022-12-19 16:17:49.840853
FAIL

======================================================================
ERROR: runTest (__main__.CythonRunTestCase)
[-1] compiling (c/cy3) and running cpdef_enums
----------------------------------------------------------------------
Traceback (most recent call last):
  File "D:\Documents\git\cython\runtests.py", line 1485, in run
    self.run_tests(result, ext_so_path)
  File "D:\Documents\git\cython\runtests.py", line 1503, in run_tests
    self.run_doctests(self.module, result, ext_so_path)
  File "D:\Documents\git\cython\runtests.py", line 1517, in run_doctests
    run_forked_test(result, run_test, self.shortDescription(), self.fork)
  File "D:\Documents\git\cython\runtests.py", line 1522, in run_forked_test
    run_func(result)
  File "D:\Documents\git\cython\runtests.py", line 1509, in run_test
    module = import_ext(module_or_name, ext_so_path)
  File "D:\Documents\git\cython\runtests.py", line 586, in import_ext
    return import_module_from_file(module_name, file_path)
  File "D:\Documents\git\cython\runtests.py", line 579, in import_module_from_file
    spec.loader.exec_module(m)
  File "<frozen importlib._bootstrap_external>", line 1184, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "<stringsource>", line 25, in init cpdef_enums (cpdef_enums.c:6088)
  File "<stringsource>", line 17, in EnumBase.__Pyx_EnumMeta.__init__ (cpdef_enums.c:3367)
TypeError: 'NoneType' object is not callable

======================================================================
ERROR: runTest (__main__.CythonRunTestCase)
[-1] compiling (cpp/cy3) and running cpdef_enums
----------------------------------------------------------------------
Traceback (most recent call last):
  File "D:\Documents\git\cython\runtests.py", line 1485, in run
    self.run_tests(result, ext_so_path)
  File "D:\Documents\git\cython\runtests.py", line 1503, in run_tests
    self.run_doctests(self.module, result, ext_so_path)
  File "D:\Documents\git\cython\runtests.py", line 1517, in run_doctests
    run_forked_test(result, run_test, self.shortDescription(), self.fork)
  File "D:\Documents\git\cython\runtests.py", line 1522, in run_forked_test
    run_func(result)
  File "D:\Documents\git\cython\runtests.py", line 1509, in run_test
    module = import_ext(module_or_name, ext_so_path)
  File "D:\Documents\git\cython\runtests.py", line 586, in import_ext
    return import_module_from_file(module_name, file_path)
  File "D:\Documents\git\cython\runtests.py", line 579, in import_module_from_file
    spec.loader.exec_module(m)
  File "<frozen importlib._bootstrap_external>", line 1184, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "<stringsource>", line 25, in init cpdef_enums (cpdef_enums.cpp:6100)
  File "<stringsource>", line 17, in EnumBase.__Pyx_EnumMeta.__init__ (cpdef_enums.cpp:3379)
TypeError: 'NoneType' object is not callable

======================================================================
FAIL: runTest (__main__.EndToEndTest)
[-1] End-to-end cpdef_enums_import
----------------------------------------------------------------------
Traceback (most recent call last):
  File "D:\Documents\git\cython\runtests.py", line 1994, in runTest
    self.assertEqual(0, res, "non-zero exit status, last output was:\n%r\n-- stdout:%s\n-- stderr:%s\n" % (
AssertionError: 0 != 1 : non-zero exit status, last output was:
'C:\\Program Files\\Python310\\python.exe -c import import_enums_test'
-- stdout:
-- stderr:Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "D:\Documents\git\cython\TEST_TMP\run\cpdef_enums_import\import_enums_test.py", line 3, in <module>
    from enums import *
  File "<stringsource>", line 25, in init enums
  File "<stringsource>", line 17, in EnumBase.__Pyx_EnumMeta.__init__
TypeError: 'NoneType' object is not callable



----------------------------------------------------------------------
Ran 3 tests in 9.065s

FAILED (failures=1, errors=2)
Most expensive pipeline stages: 'inject_utility_code_stage': 0.25 / 2 (0.124 / run), 'generate_pyx_code_stage': 0.16 / 2 (0.078 / run), 'parse': 0.03 / 4 (0.007 / run), 'AnalyseExpressionsTransform': 0.03 / 18 (0.001 / run), 'AnalyseDeclarationsTransform': 0.02 / 18 (0.001 / run), 'ControlFlowAnalysis': 0.01 / 18 (0.001 / run), 'RemoveUnreachableCode': 0.01 / 36 (0.000 / run), 'FindInvalidUseOfFusedTypes': 0.01 / 18 (0.000 / run), 'OptimizeBuiltinCalls': 0.01 / 18 (0.000 / run), 'DropRefcountingTransform': 0.01 / 18 (0.000 / run)
Times:
etoe-build  :     3.49 sec  (   1,  3.495 / run) - slowest: 'c:cpdef_enums_import(1)' (3.49s)
compile-cpp :     2.13 sec  (   1,  2.130 / run) - slowest: 'cpp:cpdef_enums' (2.13s)
compile-c   :     2.04 sec  (   1,  2.038 / run) - slowest: 'c:cpdef_enums' (2.04s)
cython      :     0.75 sec  (   2,  0.377 / run) - slowest: 'c:cpdef_enums' (0.55s), 'cpp:cpdef_enums' (0.20s)
etoe-run    :     0.04 sec  (   1,  0.043 / run) - slowest: 'c:cpdef_enums_import(2)' (0.04s)
ALL DONE
```

</details>

Reply:
OK fair enough